src/views/stadiumview.cpp

```diff
@@ -43,5 +53,244 @@ void StadiumView::resetView()
+             }
+            break;
+
+        case GREATEST_DISTANCE:
+            list = maxDistanceTeams(Database::getTeamsAndStadiums());
+            for(unsigned int index = 0; index < list.size(); ++index)
+            {
+                    totalSeatingCapacity +=  list[index].second.getSeatCap();
+            }
+            break;
+
+        case SHORTEST_DISTANCE:
+            list = minDistanceTeams(Database::getTeamsAndStadiums());
+            for(unsigned int index = 0; index < list.size(); ++index)
+            {
+                    totalSeatingCapacity +=  list[index].second.getSeatCap();
+            }
+            break;
+
+        case ALL_STADIUMS:
+            list = listT;
+            for(unsigned int index = 0; index < list.size(); ++index)
+            {
+                    totalSeatingCapacity +=  list[index].second.getSeatCap();
+            }
+            break;
+    }
+
+    //populate the widget from the list that corresponds to the button clicked
+    m_stadiumList->populateWidget(list);
+
+    //Set labels to reflect what is in list
+    ui->TotalStadiumsLabel->setText(QString("Total Amount Of Teams/Stadiums: ") + QString::fromStdString(std::to_string(list.size())));
+    ui->TotalSeatingCapacityLabel->setText(QString("Total Seating Capacity: ") + QString::fromStdString(std::to_string(totalSeatingCapacity)));
+}
+
+/**
+ * onStadiumClicked is a slot that displays the given stadium and
+ * corresponding team information inside a QMessageBox when a signal
+ * that emits stadiums id's is connected to it.
+ *
+ * @param stadiumId Id of stadium to display.
+ */
+void StadiumView::onStadiumClicked(int stadiumId)
+{
+    Stadium stadium = Database::findStadiumById(stadiumId);
+    Team team = Database::findTeamById(stadium.getTeamId());
+
+    //Don't question the spaces, it is to deal with message box not treating spaces the same width as a character
+    std::string msg;
+    msg = std::string("\tTeam Information\n")
+         +std::string("Team Name:            ") + team.getName() + '\n'
+         +std::string("League:                     ") + team.LEAGUE_STRING[team.league] + '\n'
+         +std::string("Stadium Name:       ") + stadium.getName() + '\n'
+         +std::string("Location:                  ") + stadium.getLocation() + '\n'
+         +std::string("Year Opened:           ") + std::to_string(stadium.getYearOpened()) + '\n'
+         +std::string("Seating Capacity:    ") + StadiumList::commaSeparate(std::to_string(stadium.getSeatCap())) + '\n'
+         +std::string("Typology:                 ") + stadium.TYPOLOGY_STRING[stadium.typology] + '\n'
+         +std::string("Roof Type:                ") + stadium.ROOF_STRING[stadium.roof] + '\n'
+         +std::string("Playing Surface:       ") + stadium.SURFACE_STRING[stadium.surface] + '\n'
+         +std::string("Dist To Cntr Field:    ") + StadiumList::commaSeparate(std::to_string(stadium.getCenterFieldDist())) + '\n';
+    QMessageBox box;
```

Comment:
Not sure how I feel about a pop-up window for this. Another solution would be to have a small section in the view that does the same thing as this does. This is obviously up for debate since there really isn't a good way to do this. In reality, we wouldn't even have this (friken lebo) since its redundant as all the info is already there in `StadiumList`.